zerver/lib/test_classes.py

```diff
@@ -850,3 +854,68 @@ def setUp(self) -> None:
 
     def setUpBeforeMigration(self, apps: StateApps) -> None:
         pass  # nocoverage
+
+class LDAPTestCase(ZulipTestCase):
+    """
+    Base class for LDAP tests. Takes care of the mock_ldap setup, loads
+    a directory from fixtures/ldap/directory.json with various entries to be
+    used by tests. If a test wants to specify its own directory, it can just replace
+    self.mock_ldap.directory with its own content, but in most cases it should be
+    enough to use change_user_attr to make simple modifications to the pre-loaded
+    directory. If new user entries are needed to test for some additional unusual
+    scenario, it's most likely best to add that to directory.json.
+    """
+
+    def setUp(self) -> None:
+        directory = ujson.loads(self.fixture_data("directory.json", type="ldap"))
+
+        # Load binary attributes. If in "directory", an attribute as its value
+        # has a string starting with "file:", the rest of the string is assumed
+        # to be a path to the file from which binary data should be loaded,
+        # as the actual value of the attribute in ldap.
+        for dn, attrs in directory.items():
+            for attr, value in attrs.items():
+                if isinstance(value, str) and value.startswith("file:"):
+                    with open(value[5:], 'rb') as f:
+                        attrs[attr] = [f.read(), ]
+
+        ldap_patcher = mock.patch('django_auth_ldap.config.ldap.initialize')
+        self.mock_initialize = ldap_patcher.start()
+        self.mock_ldap = MockLDAP(directory)
+        self.mock_initialize.return_value = self.mock_ldap
+
+    def tearDown(self) -> None:
+        self.mock_ldap.reset()
+        self.mock_initialize.stop()
```

Comment:
If we wanted to try not having a separate test class, I think we could do `if self.mock_ldap is not None:` for these, and then just have tests intending to use LDAP replace the `setUp` method with the test calling `self.initialize_default_ldap_database()` at the start.  

(This potentially should replace the code in ZulipLDAPAuthBackendBase that conditionally calls `init_fakeldap`, which was I think intended to do what you're trying to do here).

Reply:
I added this at the end of ZulipTestCase's tearDown. Seems like a good follow up indeed to unify things with init_fakeldap, I'll try to get that done next.